schema_based_on_diagram.sql

```diff
@@ -0,0 +1,60 @@
+CREATE DATABASE clinic_db;
+
+CREATE TABLE "patients"(
+    "id" INTEGER NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    "date_of_birth" DATE NOT NULL,
+   );
+   
+ALTER TABLE
+    "patients" ADD PRIMARY KEY("id");
+    
+CREATE TABLE "medical_histories"(
+    "id" INTEGER NOT NULL,
+    "admitted_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "patient_id" INTEGER  NOT NULL  REFERENCES patients(id),,
+    "status" VARCHAR(255) NOT NULL,
+    );
+ALTER TABLE
+    "medical_histories" ADD PRIMARY KEY("id");
+CREATE TABLE "invoices"(
+    "id" INTEGER NOT NULL,
+    "total_amount" INTEGER NOT NULL,
+    "generated_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "payed_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "medical_history_id" INTEGER NOT NULL REFERENCES medical_histories(id),
+    
+);
+ALTER TABLE
+    "invoices" ADD PRIMARY KEY("id");
+CREATE TABLE "invoice_items"(
+    "id" INTEGER NOT NULL,
+    "unit_price" DECIMAL(8, 2) NOT NULL,
+    "quantity" INTEGER NOT NULL,
+    "total_price" DECIMAL(8, 2) NOT NULL,
+    "invoice_id" INTEGER NOT NULL REFERENCES invoices(id),
+    "treatment_id" INTEGER NOT NULL REFERENCES treatments(id),
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE diagnosis(
+      treat_id INTEGER NOT NULL  REFERENCES treatments(id),
+      med_hist_id INTEGER NOT NULL   REFERENCES medical_histories(id),
+      PRIMARY KEY( treat_id, med_hist_id)
+);
+
+CREATE TABLE "treatments"(
+    "id" INTEGER NOT NULL,
+    "type" VARCHAR(255) NOT NULL,
+    "name" VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    "treatments" ADD PRIMARY KEY("id");
+ALTER TABLE
+    "medical_histories" ADD CONSTRAINT "medical_histories_patient_id_foreign" FOREIGN KEY("patient_id") REFERENCES "patients"("id");
+ALTER TABLE
+    "invoice_items" ADD CONSTRAINT "invoice_items_treatment_id_foreign" FOREIGN KEY("treatment_id") REFERENCES "treatments"("id");
+ALTER TABLE
+    "invoice_items" ADD CONSTRAINT "invoice_items_invoice_id_foreign" FOREIGN KEY("invoice_id") REFERENCES "invoices"("id");
+ALTER TABLE
+    "invoices" ADD CONSTRAINT "invoices_medical_history_id_foreign" FOREIGN KEY("medical_history_id") REFERENCES "medical_histories"("id");
```

Comment:
How are you @Nemwel-Boniface  thank you for your review , however we are a bit confused because we added  these when we altered the tables. Can you kindly assist on what exactly is required .

Reply:
Hello, @NATASHA-ct kindly double check. When you are altering the tables you are adding primary keys to the said tables and there is nowhere your team is creating indexes. I hope this helps clear the air.